File: C++/Leetcode 2545.cpp
#include <stdio.h>
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<vector<int>> sortTheStudents(vector<vector<int>>& score, int k) {
        vector<pair<int,int>> temp;
        vector<vector<int>> answer = vector<vector<int>>(score.begin(),score.end());
        
        for (int i = 0; i < score.size(); i++)
        {
            temp.push_back(make_pair(score[i][k],i));
        }
        
        sort(temp.begin(),temp.end());
        
        
        int n = temp.size();
        for (int i = 0; i < score.size(); i++)
        {
            int row = temp[n - i - 1].second;
            for (int j = 0; j < score[i].size(); j++)
            {
                answer[i][j] = score[row][j];
            }
            // score[i][k] = temp[i];
        }
        return answer;
             
    }
};

File: Python/Leetcode 2545.py
from ast import List


class Solution:
    def sortTheStudents(self, score: List[List[int]], k: int) -> List[List[int]]:
        answer = copy.deepcopy(score)
        
        temp_list = []
        
        for i in range(0,len(score)):
            list_pair = []
            list_pair.append(score[i][k])
            list_pair.append(i)
            temp_list.append(list_pair)
        
        temp_list.sort()
        # print(temp_list)
        for i in range(0,len(score)):
            row = temp_list[len(temp_list) -1 - i][1]
            # print(row)
            
            # print(score)
            for j in range(0,len(score[0])):
                answer[i][j] = score[row][j]
            # print(answer)
        
        return answer